src/System.Collections.Specialized/src/System/Collections/Specialized/OrderedDictionary.cs

```diff
@@ -334,9 +360,11 @@ public void RemoveAt(int index)
             {
                 throw new ArgumentOutOfRangeException(nameof(index));
             }
-            object key = ((DictionaryEntry)objectsArray[index]).Key;
-            objectsArray.RemoveAt(index);
-            objectsTable.Remove(key);
+            EnsureObjectsTable();
```

Comment:
If these were not initialized, then the ArgumentOutOfRangeException would have been thrown, so you do not need to initialize?

Reply:
Looks as a breaking change.
Docs 
https://docs.microsoft.com/en-us/dotnet/api/system.collections.specialized.ordereddictionary.removeat?view=netframework-4.7.2#System_Collections_Specialized_OrderedDictionary_RemoveAt_System_Int32_ 
https://docs.microsoft.com/en-us/dotnet/api/system.collections.hashtable.remove?view=netcore-2.1
don't describe the exception. And I don't see the exception in code
https://source.dot.net/#System.Runtime.Extensions/System/Collections/ArrayList.cs,ff69c0194c6935a4
https://source.dot.net/#System.Private.CoreLib/shared/System/Collections/Hashtable.cs,1031

Reply:
This looks like it already had this behavior previously but the docs hasn't been updated. 

What @danmosemsft meant here is that we don't need to call EnsureObjects apis since you're calling Count and count returns 0 when _objectArrays == null https://github.com/dotnet/corefx/pull/32001/files/551ba9e70c3e5ace47c8ccb36a0815cfc074fb3a#diff-56df9454f3e73140498b3984d3fd8ab5R84

So that means that if these were not initialized, Count will be 0, therefore index >= Count if index > 0 so it will hit: https://github.com/dotnet/corefx/pull/32001/files/551ba9e70c3e5ace47c8ccb36a0815cfc074fb3a#diff-56df9454f3e73140498b3984d3fd8ab5R361

Reply:
Thanks for clarify. I see the same in Insert() method too. Fix both?

Reply:
Yes I think it makes sense to remove it there as well.